plugins/nf-schema/src/main/nextflow/validation/Utils.groovy

```diff
@@ -52,13 +52,19 @@ public class Utils {
     }
 
     // Converts a given file to a List
-    public static List fileToList(Path file, Path schema) {
+    public static Object fileToObject(Path file, Path schema) {
         def String fileType = Utils.getFileType(file)
         def String delimiter = fileType == "csv" ? "," : fileType == "tsv" ? "\t" : null
+        def Map schemaMap = (Map) new JsonSlurper().parse( schema )
         def Map types = variableTypes(schema)
 
-        if (types.find{ it.value == "array" } as Boolean && fileType in ["csv", "tsv"]){
-            def msg = "Using \"type\": \"array\" in schema with a \".$fileType\" samplesheet is not supported\n"
+        if (schemaMap.type == "object" && fileType in ["csv", "tsv"]) {
+            def msg = "Cannot create an object from a CSV or TSV samplesheet. Please use a JSON or YAML file instead."
```

Comment:
it's not clear to me what "cannot create an object" means, from a user point of view. In which case will this error happen?

Reply:
Fair point! I'll try to think of a better way to describe this :)

Reply:
Fair point! I'll try to think of a better way to describe this :)

Reply:
@mirpedrol is this better?

Reply:
yes, thanks!